### DISCUSSION TITLE
Groups - don't select all when group-label is clicked

### DISCUSSION THREAD

**Discussion by HaloElite** [soc221b/vue-next-select#410] [General]
I got a question about groups. Is it possible to disable the select-all option when clicking the group option labes?

**Comment by soc221b** [comment]
UPDATE: this does not works, it would disabled child option too.

---

~~I think we could pass a function to `disabled-by`, and return `true` when the given option is not leaf option in that function.~~

```js
function disabledBy (option) {
    return option.isGroup ? true : false
}
```

You could wrap this feature in a component `WrappedSelect`, and extends this feature in that component.

Hope this helps you 😄 .

**Reply by HaloElite** [reply]
All right. Too bad it's not possible yet. But maybe it'll be a nice feature for the future. For now I'll go with a workaround then. ^^

**Reply by soc221b** [reply]
Sorry for that.

I found a workaround, we can use `dropdown-item` slot and pass an attribute: `@click.capture.stop` to it when the corresponding option is a group option.

```html
<vue-select ...>
    <template #dropdown-item="{ option }">
        <template v-if="option.isGroup">
            <div @click.capture.stop>{{ option.label }}</div>
        </template>
        <!-- ... -->
    </template>
</vue-select>
```

**Reply by HaloElite** [reply]
This works like a charm! Thanks for the effort. =)
